lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs

```diff
@@ -147,10 +147,11 @@ Checkpoint
     deriving Show Generic
 
 ProtocolParameters
-    protocolParametersWalletId              W.WalletId  sql=wallet_id
-    protocolParametersFeePolicy             W.FeePolicy sql=fee_policy
-    protocolParametersTxMaxSize             Word16      sql=tx_max_size
-    protocolParametersDecentralizationLevel Percentage  sql=decentralization_level
+    protocolParametersWalletId              W.WalletId      sql=wallet_id
+    protocolParametersFeePolicy             W.FeePolicy     sql=fee_policy
+    protocolParametersTxMaxSize             Word16          sql=tx_max_size
+    protocolParametersDecentralizationLevel Percentage      sql=decentralization_level
+    protocolParametersDesiredNumberOfPools  Word16 Maybe    sql=desired_pool_number
```

Comment:
Why is this a `Maybe` :thinking:  ?

Reply:
I looked at PParams in ledger for byron - there is no such a thing. So it does not make sense to talk about it (of course there is no staking for byron). So I thought Nothing there make sense. For jormungandr there is no such thing exposed so also set Nothing (although it is sensible as stake pools are there). For shelley we have this parameter in ledger `nOpt` and we can query the desired number of stake pool (which is 100 btw)

Reply:
There's `pool_participation_capping` for Jörmungandr, which has the same meaning. For Byron, we could default to 0 here, I think it'll be less misleading than having to carry a Maybe around for Shelley :/

Reply:
ok, will use Word16 instead of Maybe Word16 with 0 for Byron

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/1827/commits/3957dbcaf320cc884c13d43aef0294c7b4b8123e (also used pool participation capping max value for jormunagndr)